packages/presentation/lib/screen/login/login_view/main_login.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/material.dart';
+import 'package:presentation/core/base_bloc/bloc_state.dart';
+import 'package:presentation/core/base_bloc/stream_platform_stack_content.dart';
+import 'package:presentation/screen/home/home_tab_bar.dart';
+import 'package:presentation/screen/login/bloc/login_bloc.dart';
+import 'package:presentation/screen/login/bloc/login_data.dart';
+import 'package:presentation/screen/login/login_view/login_key.dart';
+
+class LoginPage extends StatefulWidget {
+  const LoginPage({Key? key}) : super(key: key);
+
+  @override
+  State<LoginPage> createState() => _LoginPageState();
+}
+
+class _LoginPageState extends BlocState<LoginPage, LoginBloc> {
+  @override
+  void initState() {
+    super.initState();
+    bloc.initState();
+  }
+
+  @override
+  void dispose() {
+    bloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamPlatformStackContent(
+      dataStream: bloc.dataStream,
+      children: (blocData) {
+        if (blocData.isLoading) {
+          return const CircularProgressIndicator.adaptive();
+        } else {
+          final screenData = blocData.data;
+          if (screenData is LoginData) {
+            if (screenData.exception != null) {
```

Comment:
In My point of view , code that contains a lot of  nested "if" structure - it's unreadable, difficult to understand and incomprehensible code.
Please try use structure with out nested "if".

Reply:
it depends on your condition which you want get
I think 2 conditions are ok
if there are more of course it can effect to performance